parteh/PRTAllometricCarbonMod.F90

```diff
@@ -862,6 +884,565 @@ subroutine DailyPRTAllometricCarbon(this)
+    !       as possible.  This also includes the storage pool, even though carbon may
+    !       be drawn from the storage.
+    ! -----------------------------------------------------------------------------------
+
+    ! Identify living organs (plus storage) that are under target.  Priority is given
+    ! to the organs (or storage) that are the most depleted, without a pre-determined
+    ! sequence.
+    leaf_below_target   = max( 0.0_r8, target_leaf_c   - sum(leaf_c(1:nleafage)))
+    fnrt_below_target   = max( 0.0_r8, target_fnrt_c   - fnrt_c   )
+    sapw_below_target   = max( 0.0_r8, target_sapw_c   - sapw_c   )
+    store_below_target  = max( 0.0_r8, target_store_c  - store_c  )
+    struct_below_target = max( 0.0_r8, target_struct_c - struct_c )
+    total_below_target  = leaf_below_target + fnrt_below_target  + sapw_below_target + &
+                          store_below_target + struct_below_target
+
+    replenish_allom_check: if ( total_below_target > nearzero ) then
+       ! Available carbon for transfer is the sum of stored carbon and the daily
+       ! carbon balance.
+       available_carbon = store_c + carbon_balance
+
+       ! Scale flux so pools can be replenish simultaneously.
+       leaf_c_flux   = min( leaf_below_target, &
+                            available_carbon * leaf_below_target   / total_below_target )
+       fnrt_c_flux   = min( fnrt_below_target, &
+                            available_carbon * fnrt_below_target   / total_below_target )
+       sapw_c_flux   = min( sapw_below_target, &
+                            available_carbon * sapw_below_target   / total_below_target )
+       store_c_flux  = min( store_below_target, &
+                            available_carbon * store_below_target  / total_below_target )
+       struct_c_flux = min( struct_below_target, &
+                            available_carbon * struct_below_target / total_below_target )
+
+       ! Replenish pools
+       leaf_c(iexp_leaf) = leaf_c(iexp_leaf) + leaf_c_flux
+       fnrt_c            = fnrt_c            + fnrt_c_flux
+       sapw_c            = sapw_c            + sapw_c_flux
+       store_c           = store_c           + store_c_flux
+       struct_c          = struct_c          + struct_c_flux
+
+       ! Update carbon balance
+       carbon_balance    = carbon_balance - &
+                           ( leaf_c_flux + fnrt_c_flux + sapw_c_flux + &
+                             store_c_flux + struct_c_flux )
+
+    end if replenish_allom_check
+
+
+    ! -----------------------------------------------------------------------------------
+    ! IV.  If carbon balance is negative, reduce the storage pool.  Otherwise, try to
+    !      fill the storage pool before growing.
+    ! -----------------------------------------------------------------------------------
+    update_storage: if ( carbon_balance < 0.0_r8 .or. is_hydecid_dormant ) then
```

Comment:
@mpaiao could you provide some text here on why we perform this flux for hydro deciduous dormant?